Crashlytics/Crashlytics/Models/Record/FIRCLSRecordAdapter.m

```diff
@@ -20,21 +20,40 @@
 #import "FIRCLSInternalReport.h"
 #import "FIRCLSLogger.h"
 
+#import "FIRCLSUserLogging.h"
+
+#import <nanopb/pb.h>
+#import <nanopb/pb_decode.h>
+#import <nanopb/pb_encode.h>
+
+// [TODO]: Rename to FIRCLSReportAdapter
 @implementation FIRCLSRecordAdapter
 
-- (instancetype)initWithPath:(NSString *)folderPath {
+- (instancetype)initWithPath:(NSString *)folderPath withGoogleAppId:googleAppID {
   self = [super init];
   if (self) {
     _folderPath = folderPath;
+    _googleAppID = googleAppID;
 
     [self loadBinaryImagesFile];
     [self loadMetaDataFile];
     [self loadSignalFile];
     [self loadKeyValuesFile];
+      
+    _report = [self protoReport];
   }
   return self;
 }
 
+- (void)dealloc {
+  // [TODO]: Have to manually traverse this struct to free up memory?
```

Comment:
@mikehaney24 - Our report very struct is nested. Is calling pb_release on the top level suffice? If not, what would you recommend for tracking all the memory that needs to be released?

Reply:
AFAICT, pb_release should handle this: https://jpa.kapsi.fi/nanopb/docs/reference.html#pb-release

I'd verify with the profiler, just in case.

Reply:
You're not going to do more than a few percent better than `pb_release` by writing this by hand. Meanwhile you're very likely to miss something and leak--perhaps not now but when you add something to the proto later. Just use `pb_release`.

Reply:
Gotcha, thanks for the input guys!